server/server.go

```diff
@@ -279,19 +293,37 @@ func (s *Server) concurrentRequest(ctx context.Context, conf map[string]lookout.
 			cs, err := send(ctx, a.Client, settings)
 			if err != nil {
 				aLogger.Errorf(err, "analysis failed")
+				if s.ExitOnError {
+					errCh <- err
+				}
 				return
 			}
 
 			if len(cs) == 0 {
 				aLogger.Infof("no comments were produced")
+				return
 			}
 
-			comments.Add(a.Config, cs...)
+			result = &lookout.AnalyzerComments{
+				Config:   a.Config,
+				Comments: cs,
+			}
 		}(name, a)
 	}
-	wg.Wait()
 
-	return comments.Get()
+	var comments []lookout.AnalyzerComments
+	for i := 0; i < len(s.analyzers); i++ {
+		select {
+		case err := <-errCh:
+			return nil, err
```

Comment:
Should we do a context cancellation for all the other `send` still running in goroutines?

Reply:
I thought about it and most probably we don't need to cancel it here.
We will propagate the error to the cmd which would catch it and cancel everything.